src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/config/WebClientConfiguration.kt

```diff
@@ -0,0 +1,86 @@
+package uk.gov.justice.digital.hmpps.hmppsregisterstonomisupdate.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.reactive.function.client.ExchangeStrategies
+import org.springframework.web.reactive.function.client.WebClient
+
+@Configuration
+class WebClientConfiguration(
+  @Value("\${oauth.endpoint.url}") private val oauthRootUri: String,
+  @Value("\${prison.endpoint.url}") private val prisonRootUri: String,
+  @Value("\${court.register.endpoint.url}") private val courtRegisterRootUri: String,
+) {
+
+  @Bean
+  fun prisonApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("prison-api")
+
+    return WebClient.builder()
+      .baseUrl(prisonRootUri)
+      .apply(oauth2Client.oauth2Configuration())
+      .exchangeStrategies(
+        ExchangeStrategies.builder()
+          .codecs { configurer ->
+            configurer.defaultCodecs()
+              .maxInMemorySize(-1)
```

Comment:
I think we tend to manage this with configuration nowadays: https://github.com/ministryofjustice/hmpps-template-kotlin/blob/main/src/main/resources/application.yml#L9